### DISCUSSION TITLE
@required blade directive

### DISCUSSION THREAD

**Discussion by GuidoBelluomo** [laravel/framework#43102] [Ideas] [answered]
Similar to `@checked` and `@selected`, it would be very good to have a `@required` directive out of the box.

I'm currently defining it like this:
```php
Blade::directive('required', function ($expression) {
    return "<?php echo ($expression ? 'required' : ''); ?>";
});
```
It's not that much trouble, but I think it would be a welcome change. I can't figure out where directives are defined in the framework or I would have submitted a pull request for it myself

**Comment by erikn69** [comment, marked as answer]
It should be something like this
```php
Blade::directive('required', function ($condition) {
     return "<?php if({$condition}): echo 'required'; endif; ?>";
});
/* more generic 
 @echoif('required', $condition) 
 @echoif('readonly', $condition)
 @echoif('data-value="true"', $condition) */
Blade::directive('echoif', function($arguments){ 
    $args = explode(',', $arguments), 2); 
    return "<?php if({$args[1]}): e({$args[0]}); endif; ?>";
});
```
Just put that on your `ViewServiceProvider`

**Reply by erikn69** [reply]
Mark this as answered 

- #43103

**Reply by GuidoBelluomo** [reply]
You're right, I just realized it now when it wasn't working as intended that I haven't done it properly. Either way suggestion still stands!
